### PR TITLE
Improve Naga Guard line

### PR DESCRIPTION
**Pull request by MechanicalDragon963**
This PR based on issue https://github.com/wesnoth/wesnoth/issues/8992 
All stats already discussed with @Dalas121

### CODE REVIEW

**Comment by Dalas121** [comment]
For the record, I dislike the name "absorb" as well as having a varying {AMT} effect. However, I feel that changing either at this point would risk completely breaking any UMC that uses Absorb.

**Comment by Mirion147** [comment]
Level 1 seems very overpriced.

-2mp from other naga
Slightly better defense in castle/ village hexes (10% and 20% respectively)
5 more hp than fighter but 3 more xp than fighter.

Same damage as fighter but impact over blade, and only 2 chances to hit

3 more gold than fighter. 

I think the 2 mp is being overlooked here. I think this is closer to 13g, maybe even 12.

Look at Troll Whelp.

20% blade and pierce resist, 1 more damage per strike, 4 more hp, 4 less xp, regenerates. And only 1 less mp. 13 gold.

Things get slightly better for level 2 and 3 because of the absorb ability. I think it's an interesting system for the Unwieldy attack and absorb defense. I don't really buy the description in reference to ranged attacks, but that's not worth the argument imo, I think there are ways to explain the result so its fine for now. 

That being said, still down 2mp, +4-7hp. The xp is fair. It's a pretty massive jump in damage, reduced incoming attack damage, and but also still no ranged attack.
Looking at other naga, Troll, even orcish grunt, I think 25 gold actually works fine, 28 is too high. This unit is not on the same level as Netcaster, Swordsman, Shock Trooper, or Deulist (26-29 gold) 

Then 50 gold for level 3 is actually fairly good, though I would say that forcing a level 3 to be susceptible to all ranged attacks for free and only be able to deal 22 retaliatory damage sans resists is a bit wild. It's a bit of a rough situation that the unit has wound up in. It seems like the shield attack would need to be raised to 14x2, and reduce the mace attack to 12x3

Naga Guard line suggestions:
Level 1: 17g > 13(12)g

Level 2: 28g >25g

Level 3: mace 13x3 > 12x3, shield 11x2 > 14x2

And if Sicarius is going to lose deflect then it also needs to be addressed. It seems pointless to remove it in this PR but then leave it directly needing attention.

**Comment by Dalas121** [comment]
> And if Sicarius is going to lose deflect then it also needs to be addressed. It seems pointless to remove it in this PR but then leave it directly needing attention.

Sicarius is my fault. I'd asked for deflect's removal with the expectation of a compensating buff, but hadn't actually asked for any specific buff. My current thought on the Sicarius are in #9532

I have no strong feelings about any of the other points, though do note that a Troll Whelp comparison may not be 
accurate given water units' 60% defense - compare the Merman Fighter to a Spearman, for example.

Absorb is weird and doesn't really have any precedent in mainline, so I have a hard time judging its balance without a lot of playtesting (which I believe Mech has done but I haven't).

**Comment by MechanicalDragon963** [comment]
Lowered Guard's cost to 16. I think now his price is fair enough.

**Comment by Mirion147** [comment]
> Sicarius is my fault. I'd asked for deflect's removal with the expectation of a compensating buff, but hadn't actually asked for any specific buff. My current thought on the Sicarius are in #9532

Yea that's fair, I would say just moving everything to that PR would be for the best regarding the Sicarius. Unless there's a reason it should remain here? 

> I have no strong feelings about any of the other points, though do note that a Troll Whelp comparison may not be 
> accurate given water units' 60% defense - compare the Merman Fighter to a Spearman, for example.

I agree that the troll is going to have worse defense stats (generally speaking) and less areas of reachability. But the fact that the troll is better in many areas and the guard is better in others makes them comparable. My point is that they both have nearly equal strengths and weaknesses but the guard costs 4 (now 3) gold more. 


> Lowered Guard's cost to 16. I think now his price is fair enough. 

Alright, might I ask why you think 16 is fair? What is the basis for such a high price?

**Comment by ForestDragon-wesnoth** [comment]
I like the naga rework. now their shield is finally consistent and useful more often

As for level 3 mace damage, the PR's current 13x3 seems reasonable. 12x3 would be to weak for a lvl3, especially considering it is attack-only

**Comment by Mirion147** [comment]
> I like the naga rework. now their shield is finally consistent and useful more often
> 
> As for level 3 mace damage, the PR's current 13x3 seems reasonable. 12x3 would be to weak for a lvl3, especially considering it is attack-only

I like the rework as well, but the gold for level 1 and and 2 are off.

As for level 3, I don't generally have an issue with it, the problem is the shield damage. Restricting a level 3 unit to only doing a max of 22 retaliatory damage while also not having a ranged attack at all means they're just going to get decimated. I understand that absorb will help with some of the damage from melee, but there's not enough of a deterrent. It will always be worth it to attack, making the absorb actually somewhat not worth it for the unit. 

That's why I suggested increasing the shield damage, at the cost of the mace damage. I suppose you could reduce the hp, but it only increases by 10 from level 2 already. There's not a lot of wiggle room. You could increase the gold by 3, that would also be sufficient

**Comment by MechanicalDragon963** [comment]
To be honest, I don't think big theorizing is appropriate for a unit that is not used in any way in the mainline at the moment. Even on the github in a few months (I first brought it up in June) no one but Dalas has said anything on the topic. Moreover, I don't remember UMCs in which Naga Guardian can be recruited (if you know such UMCs - let me know, I'll try to contact their authors). 

I have a different approach for this situation. I plan to test my changes in the VS campaign (both by playing myself and by watching dozens of replays from players). If there is something wrong with the Naga Guardian line based on the results of the tests - I will be the first to know about it long before 1.20 and make changes. 

So for now, I'm asking @Dalas121  for approval of the changes.

**Comment by rodolphecombe** [comment]
The modification in the description of absorb do not correspond to its effect as AMT is in multiplier of damage inflicted while the definition now would say that it is a percentage of damage absorbed. 
And as was stated by Soliton, we do not want variable translatable string in mainline. 
Thus PR #9520

**Comment by igorbat** [comment]
hi here! I see an intention to fill a somewhat elite-fish-defender role with 2 strikes in melee (lvl1) while keeping 0% resists (due to absorb ability on the higher levels)
From MP perspective on the current duel maps, 
I would suggest following changes to the current stats of lvl1:

- 8-2 -> 9-2, making them somewhat grunt level of melee dmg

- 5 mp -> 6 mp: this would allow guard to move 3 tiles per turn on flat, and would allow guard to take water villages which are usually (6 tiles away from the keep by water tiles ) / (4 water + 1 flat). 6 mp is still less than 7mp which other nagas have. And this would be still faster than Merman Hoplite. This way we would underline the meaningful mobility of nagas

- 2mp -> 1 mp castle movement cost. This is the guard!

- 38hp -> 40hp. Lets compensate the lack of resists and underline the defender nature.

===

As for Lvl2 a Lvl3 3, you could keep movement changes for lvl1 and buff hp a bit (48->50, 58->60), but I don't insist here at all.

**Comment by Mirion147** [comment]
> From MP perspective on the current duel maps, 
> I would suggest following changes to the current stats of lvl1:
> 
> - 8-2 -> 9-2, making them somewhat grunt level of melee dmg
> 
> - 5 mp -> 6 mp: this would allow guard to move 3 tiles per turn on flat, and would allow guard to take water villages which are usually (6 tiles away from the keep by water tiles ) / (4 water + 1 flat). 6 mp is still less than 7mp which other nagas have. And this would be still faster than Merman Hoplite. This way we would underline the meaningful mobility of nagas
> 
> - 2mp -> 1 mp castle movement cost. This is the guard!
> 
> - 38hp -> 40hp. Lets compensate the lack of resists and underline the defender nature.

I think these changes would help the kennel 1 guard be worth the current 16 gold, though I do think an argument could put them at 15g, I think 16g is within reason

> As for Lvl2 a Lvl3 3, you could keep movement changes for lvl1 and buff hp a bit (48->50, 58->60), but I don't insist here at all. 

I agree that this would help, but I would still push for higher shield damage for the level 3. Regardless of the shield damage, the changes mentioned above would need an increase in cost, maybe 54g or 57g with the increased damage

**Comment by rodolphecombe** [comment]
> For the record, I dislike the name "absorb" as well as having a varying {AMT} effect. However, I feel that changing either at this point would risk completely breaking any UMC that uses Absorb - thus the implementation in this PR.

It just needs to be deprecated and replaced for those two units. It will be removed entirely in one or two stable versions

**Comment by Dalas121** [comment]
Mech, I'm about to change some of your stats in the PR. Please let me know if any of these changes are problematic, and feel free to revert them when next you're here.


> 8-2 -> 9-2, making them somewhat grunt level of melee dmg
> 38hp -> 40hp. Lets compensate the lack of resists and underline the defender nature.
> 5 mp -> 6 mp: this would allow guard to move 3 tiles per turn on flat,

Ok, will do. I'll also need to increase the L2's shield bash to 9-2, or else we run into RIPLIB issues. I believe that both 6mp and 1mp castle is overkill, so I've gone with just the mp.


> Restricting a level 3 unit to only doing a max of 22 retaliatory damage while also not having a ranged attack at all means they're just going to get decimated.

I disagree. As a level 3 the High Guard is most likely to be used by the player against the AI, and in my experience the AI rarely refrains from attacking a target within reach. Thus, I don't believe that retaliatory damage will significantly increase the High Guard's survivability vs the AI.

Compare the High Guard to the Royal Guard. If I attack the Royal Guard in melee, I might deal 30 damage and take 44 in retaliation. If I attack the High Guard in melee, I might deal 15 damage and take 22 in retaliation. I don't see a survivability gap there.

In addition, the 5mp->6mp buff will hopefully help strengthen the High Guard. If it needs more, perhaps reverting their mace damage from 13->15 (its current 1.18 value) might help compensate for Unwieldy and make the High Guard more threatening against potential ranged attackers.


> It just needs to be deprecated and replaced for those two units. It will be removed entirely in one or two stable versions

I support this. Mirion suggested "Guard" as the name for a new special - does anybody object?

**Comment by MechanicalDragon963** [comment]
In fact, igorbat's feedback has given me a lot of food for thought. I agree, I didn't quite get the cost and strength of the guardian naga right the first time. 
 
 @Dalas121 thanks for making some of the changes suggested by igorbat - I was going to do that myself, but you beat me to it.  
 
As for the absorb ability - I agree to any changes to it, as long as the principle of blocking 50% damage for shield-bearing Nagas is retained.

**Comment by soliton-** [comment]
> For the record, I dislike the name "absorb" as well as having a varying {AMT} effect. However, I feel that changing either at this point would risk completely breaking any UMC that uses Absorb - thus the implementation in this PR.

Is any UMC actually using absorb?

File: data/core/macros/weapon_specials.cfg [hunk 1]
```diff
@@ -151,7 +161,7 @@
     [damage]
         id=absorb
         name= _ "absorb"
-        description= _ "This attack puts the unit in good defensive position, and it absorbs some of the damage dealt by an enemy strike."
+        description= _ "This attack puts the unit in good defensive position, and it absorbs"+" {ATM}%"+ _" damage dealt by an enemy strike."
```
Comment (soliton-) [inline data/core/macros/weapon_specials.cfg:164]: Is there any precedence for this in mainline? I don’t think we want to subject translations to this.

**Comment by Dalas121** [comment]
> Is any UMC actually using absorb?

In #9532, Knyghtmare mentioned you had a way to grep the add-on servers and check - is that correct? I pinged you yesterday in the discord about it.

Regardless, I'm in favor of @rodolphecombe's suggestion to deprecate Absorb and Deflect, then creating a new special with a proper name like Guard or Block (or even overwriting Deflect). The new special would function identically to `{WEAPON_SPECIAL_ABSORB 0.50}`, halving incoming damage.

Since Hejnewar has been inactive for a while and his approval is necessary for most changes, there's probably no rush to close things out.

**Comment by rodolphecombe** [comment]
> > It just needs to be deprecated and replaced for those two units. It will be removed entirely in one or two stable versions
> 
> I support this. Mirion suggested "Guard" as the name for a new special - does anybody object?

I am not sure I support having the same ability for both units as it is not the case now, but I will not fight to have one that absorb 1/4 and one 1/2. 
I'm fine with Guard for the name, although I think absorb was fine too.

**Comment by Dalas121** [comment]
> I am not sure I support having the same ability for both units as it is not the case now, but I will not fight to have one that absorb 1/4 and one 1/2.

I guess I feel it's overcomplicating things to have multiple specials with identical mechanics but different values. For example, we don't have a "Magical 70%" and a "Magical 50%", nor a "Charge 2x" and a "Charge 1.5x".

And since Guard would only be used on two units right now, I feel it's especially unnecessary to have two separate specials - I'd rather have one special and use it more.

> I'm fine with Guard for the name, although I think absorb was fine too.

Ok; Guard it is then. What's the process for deprecating Absorb/Deflect?

**Comment by Dalas121** [comment]
Yeah, sounds like we can't remove/change either of those specials.

Does everything look correct with this PR right now?

**Comment by MechanicalDragon963** [comment]
> Yeah, sounds like we can't remove/change either of those specials. I've deprecated Absorb and replaced it with Guard (Deflect deprecation is in the Sicarius PR).
> 
> Does everything look correct with this PR now?

All looks fine for me.

**Comment by soliton-** [comment]
I'd wait for the checks to run through but looks good to me as well. (Don't forget to squash merge.)

**Pull request merged**